webapp/src/ts/modules/contacts/contacts.component.ts

```diff
@@ -310,6 +310,52 @@ export class ContactsComponent implements OnInit, AfterViewInit, OnDestroy {
     return this.sortDirection === 'last_visited_date';
   }
 
+  private getContactsDocIds(contactsList, usersHomePlace): string[] {
+    const docIds = contactsList.map((item) => item._id);
+    if (usersHomePlace && !docIds.includes(usersHomePlace._id)) {
+      docIds.push(usersHomePlace._id);
+    }
+    return docIds;
+  }
+
+  private findHomeIndex(updatedContacts: any[]): number {
+    return _findIndex(updatedContacts, contact => contact._id === this.usersHomePlace._id);
+  }
+
+  private checkAdditionalListItem(homeIndex: number) {
+    this.additionalListItem =
+      !this.filters.search &&
+      (this.additionalListItem || !this.appending) &&
+      homeIndex === -1;
+  }
+
+  private moveUserHomePlaceToTop(updatedContacts: any[], homeIndex: number) {
+    if (homeIndex < 0) {
+      return updatedContacts;
```

Comment:
If homeIndex is -1 (not found) and not appending, should it add it to `updatedContact`?